src/deluge/model/instrument/non_audio_instrument.cpp

```diff
@@ -34,35 +34,50 @@ void NonAudioInstrument::renderOutput(ModelStack* modelStack, StereoSample* star
 	if (activeClip) {
 		InstrumentClip* activeInstrumentClip = (InstrumentClip*)activeClip;
 
+		uint32_t rhythm = activeInstrumentClip->arpeggiatorRhythm;
+		uint32_t sequenceLength = activeInstrumentClip->arpeggiatorSequenceLength;
+		uint32_t chordPolyphony = activeInstrumentClip->arpeggiatorChordPolyphony;
+		uint32_t ratchetAmount = activeInstrumentClip->arpeggiatorRatchetAmount;
+		uint32_t noteProbability = activeInstrumentClip->arpeggiatorNoteProbability;
+		uint32_t bassProbability = activeInstrumentClip->arpeggiatorBassProbability;
+		uint32_t chordProbability = activeInstrumentClip->arpeggiatorChordProbability;
+		uint32_t ratchetProbability = activeInstrumentClip->arpeggiatorRatchetProbability;
+		uint32_t spreadVelocity = activeInstrumentClip->arpeggiatorSpreadVelocity;
+		uint32_t spreadGate = activeInstrumentClip->arpeggiatorSpreadGate;
+		uint32_t spreadOctave = activeInstrumentClip->arpeggiatorSpreadOctave;
+
+		arpeggiator.updateParams(rhythm, sequenceLength, chordPolyphony, ratchetAmount, noteProbability,
+		                         bassProbability, chordProbability, ratchetProbability, spreadVelocity, spreadGate,
+		                         spreadOctave);
+
```

Comment:
Seeing this block copied-but-not a few times makes me think that there should be a class like `ArpeggiatorConfiguration` that contains all these state variables, so it can be shared between `InstrumentClip`, `NonAudioDrum`, etc.

That would let us unify the serialization/deserialization code, avoid these 15-line-long copy blocks (since the signature for `Arpeggiator::updateParams` could just be `updateParams(ArpeggiatorConfiguration const & config)`, and make adding new config/params even easier.